packages/js/src/decorator/gutenberg.js

```diff
@@ -325,12 +320,75 @@ function getAnnotationsForBlockAttribute( attribute, block, marks ) {
 			return {
 				...annotation,
 				block: block.clientId,
-				richTextIdentifier: attributeKey,
+				richTextIdentifier: richTextIdentifier,
 			};
 		} );
 	} ) );
 }
 
+/**
+ * Gets the annotations for Yoast FAQ block and Yoast How-To block.
+ *
+ * @param {Object} attribute The attribute to apply annotations to.
+ * @param {Object} block     The block information in the state.
+ * @param {Array}  marks     The marks to turn into annotations.
+ *
+ * @returns {Array} An array of annotations for Yoast blocks.
+ */
+function getAnnotationsForYoastBlocks( attribute, block, marks ) {
+	const attributeKey = attribute.key;
+
+	// For Yoast FAQ and HowTo blocks, we create a separate annotation objects for each individual Rich Text
+
+	const { attributes: blockAttributes } = block;
+	const annotableTexts = blockAttributes[ attributeKey ];
+	// For each rich text of the attribute value, create annotations.
+	if ( block.name === "yoast/faq-block" && annotableTexts.length !== 0 ) {
+		const annotations = annotableTexts.map( item => {
+			const identifierQuestion = `${ item.id }-question`;
+			const identifierAnswer = `${ item.id }-answer`;
+
+			const annotationsFromQuestion = createAnnotations( item.jsonQuestion, identifierQuestion, attribute, block, marks );
+			const annotationsFromAnswer = createAnnotations( item.jsonAnswer, identifierAnswer, attribute, block, marks );
+
+			return annotationsFromQuestion.concat( annotationsFromAnswer );
+		} );
+
+		return flatMap( annotations );
+	}
```

Comment:
The approach for Yoast FAQ block is as follows:
* If the block is Yoast FAQ, we retrieve the `questions` attribute of the block.
   * `questions` contains an array of objects of question and answer pair, including the `id` (see screenshot for reference)
<img width="474" alt="Screenshot 2023-01-12 at 14 42 25" src="https://user-images.githubusercontent.com/48715883/212295797-2eca504a-b0e4-4bf6-8e8a-54e99b088201.png">

   * Inside the object, we create the annotations for each annotable text (rich text), in this case they are the question and the answer.
   * In addition, for each annotation, we also attach the corresponding `richTextIdentifier`
       * The `richTextIdentifier` for question is the `id` of the question-answer pair + "-question" 
       * The `richTextIdentifier` for answer is the `id` of the question-answer pair + "-answer" 
       * The format of the `richTextIdentifier` should match the `identifier` we pass in `Question.js` component

Reply:
Thank you for such a clear elaboration @FAMarfuaty  :)